src/components/node-list/node-list-tree.js

```diff
@@ -162,10 +170,20 @@ const TreeListProvider = ({
   const renderTree = (tree, modularPipelineID) => {
     // current tree node to render
     const node = tree[modularPipelineID];
+
     if (!node) {
       return;
     }
 
+    // If all children's data are empty, the subtree rooted at this node will not be rendered.
+    // in scenarios where the pipeline is being sliced, and some modular pipelines trees do not have any children
+    const allChildrenDataEmpty = node.children.every(
+      (child) => Object.keys(child.data).length === 0
+    );
+    if (allChildrenDataEmpty) {
+      return;
+    }
+
     // render each child of the tree node first
     const children = sortBy(
       node.children,
```

Comment:
So we are checking if allChildren of modular pipelines are empty. If so, we return. Will there be a case where some children are empty ? If so, the line at 173 has `child.data.name` which will error out.

Reply:
i think it will return undefined if theres no name under child.data. But good point, I'm surprised that it didn't show any error. Should we add extra check there?

Reply:
I think so may be child?.data?.name